src/Features/Core/Portable/ExtractMethod/MethodExtractor.CodeGenerator.cs

```diff
@@ -83,97 +80,93 @@ protected CodeGenerator(InsertionPoint insertionPoint, SelectionResult selection
 
-                SyntaxNode destination, newContainer;
-                if (LocalFunction)
-                {
-                    destination = InsertionPoint.With(callSiteDocument).GetContext();
+                var rootWithUpdatedCallSite = this.SemanticDocument.Root.ReplaceNode(
+                    outermostCallSiteContainer,
+                    await GenerateBodyForCallSiteContainerAsync(outermostCallSiteContainer, cancellationToken).ConfigureAwait(false));
 
-                    // No valid location to insert the new method call.
-                    if (destination == null)
-                    {
-                        return await CreateGeneratedCodeAsync(
-                            OperationStatus.NoValidLocationToInsertMethodCall, callSiteDocument, cancellationToken).ConfigureAwait(false);
-                    }
+                // Then insert the local-function/method into the updated document that contains the updated callsite.
+                var documentWithUpdatedCallSite = await this.SemanticDocument.WithSyntaxRootAsync(rootWithUpdatedCallSite, cancellationToken).ConfigureAwait(false);
+                var finalRoot = LocalFunction
+                    ? InsertLocalFunction()
+                    : InsertNormalMethod();
 
+                return await documentWithUpdatedCallSite.WithSyntaxRootAsync(finalRoot, cancellationToken).ConfigureAwait(false);
+
+                SyntaxNode InsertLocalFunction()
+                {
+                    // Now, insert the local function.
                     var info = codeGenerationService.GetInfo(
-                        new CodeGenerationContext(
-                            generateDefaultAccessibility: false,
-                            generateMethodBodies: true),
+                        new CodeGenerationContext(generateDefaultAccessibility: false),
                         Options,
-                        callSiteDocument.Project.ParseOptions);
+                        document.Project.ParseOptions);
+
+                    var localMethod = codeGenerationService.CreateMethodDeclaration(newMethodDefinition, CodeGenerationDestination.Unspecified, info, cancellationToken);
+
+                    // Find the destination for the local function after the callsite has been fixed up.
+                    var destination = InsertionPoint.With(documentWithUpdatedCallSite).GetContext();
+                    var updatedDestination = codeGenerationService.AddStatements(destination, new[] { localMethod }, info, cancellationToken);
 
-                    var localMethod = codeGenerationService.CreateMethodDeclaration(result.Data, CodeGenerationDestination.Unspecified, info, cancellationToken);
-                    newContainer = codeGenerationService.AddStatements(destination, new[] { localMethod }, info, cancellationToken);
+                    var finalRoot = documentWithUpdatedCallSite.Root.ReplaceNode(destination, updatedDestination);
+                    return finalRoot;
```

Comment:
inserting a local function is actually the easier task of the two.  we already determined the container earlier to add teh local function statement to, so we just need to find it and add to it.